### PR TITLE
Reduce tailwatcher argument

### PR DESCRIPTION
**Pull request by ganmacs**
**Which issue(s) this PR fixes**: 
no

**What this PR does / why we need it**: 

Reduce `TailWatcher`'s arguments. it's hard to maintain now.

**Docs Changes**:

no

**Release Note**: 

no

### CODE REVIEW

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -384,6 +398,9 @@ def close_watcher_handles
 
     # refresh_watchers calls @tails.keys so we don't use stop_watcher -> start_watcher sequence for safety.
     def update_watcher(path, pe)
+      log_msg = "detected rotation of #{path}"
+      log_msg << "; waiting #{@rotate_wait} seconds" # wait rotate_time if previous file exists
+
```
Comment (cosmo0920) [inline lib/fluent/plugin/in_tail.rb]: Missing `log.info` here?

Comment (ganmacs) [reply lib/fluent/plugin/in_tail.rb]: Good catch…thank you 🙏  https://github.com/fluent/fluentd/pull/2832/commits/994086913afb32f6697d8960c840a9458e63009b

File: lib/fluent/plugin/in_tail.rb [hunk 2]
```diff
@@ -384,6 +398,10 @@ def close_watcher_handles
 
     # refresh_watchers calls @tails.keys so we don't use stop_watcher -> start_watcher sequence for safety.
     def update_watcher(path, pe)
+      log_msg = "detected rotation of #{path}"
+      log_msg << "; waiting #{@rotate_wait} seconds" # wait rotate_time if previous file exists
+      log.info(log_msg)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb]: Can combine 3 lines.

File: lib/fluent/plugin/in_tail.rb [hunk 3]
```diff
@@ -655,18 +692,21 @@ def on_rotate(stat)
...
 
-          log_msg = "detected rotation of #{@path}"
-          log_msg << "; waiting #{@rotate_wait} seconds" if watcher_needs_update # wait rotate_time if previous file exists
-          @log.info log_msg
-
           if watcher_needs_update
             @update_watcher.call(@path, swap_state(@pe))
           else
-            @io_handler = IOHandler.new(self, &method(:wrap_receive_lines))
+            @log.info "detected rotation of #{@path}"
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb]: Is this line needed?

Comment (ganmacs) [reply lib/fluent/plugin/in_tail.rb]: (L658)[https://github.com/fluent/fluentd/pull/2832/files#diff-1da710c9dcc8d0fc57996df7a9d39695L658] and (L660)[https://github.com/fluent/fluentd/pull/2832/files#diff-1da710c9dcc8d0fc57996df7a9d39695L660] is invoked even if `watcher_needs_update` is `false`, right?

File: lib/fluent/plugin/in_tail.rb [hunk 4]
```diff
@@ -524,59 +544,76 @@ def parse_multilines(lines, tail_watcher)
...
+      def initialize(path, log, &callback)
+        @callback = callback
+        @log = log
+        super(path)
+      end
+
+      def on_change(prev, cur)
+        @callback.call
+      rescue
+        # TODO log?
```
[the remark this answers is not in the document]
Comment (ganmacs) [reply lib/fluent/plugin/in_tail.rb]: fixed https://github.com/fluent/fluentd/pull/2832/commits/269ea68086c82e34ae0423c86bc89dcc5f85fc67

File: lib/fluent/plugin/in_tail.rb (hunk 2 above)
Comment (ganmacs) [reply lib/fluent/plugin/in_tail.rb]: fixed https://github.com/fluent/fluentd/pull/2832/commits/857d9cb305b8b59a1bd45e8198f597a633718986

File: lib/fluent/plugin/in_tail.rb (hunk 3 above)
Comment (repeatedly) [reply lib/fluent/plugin/in_tail.rb]: Ah, I see.

**Pull request merged**